ludwig/benchmarking/reporting.py

```diff
@@ -33,3 +39,92 @@ def create_metrics_report(experiment_name: str) -> Tuple[Dict[str, Any], str]:
+    run_usage_info["self_cpu_time_total"] = function_event.self_cpu_time_total
+    run_usage_info["self_cpu_time_total_str"] = function_event.self_cpu_time_total_str
+    run_usage_info["cuda_time_total"] = function_event.cuda_time_total
+    run_usage_info["cuda_time_total_str"] = function_event.cuda_time_total_str
+    run_usage_info["self_cuda_time_total"] = function_event.self_cuda_time_total
+    run_usage_info["self_cuda_time_total_str"] = function_event.self_cuda_time_total_str
+    run_usage_info["cpu_time_total"] = function_event.cpu_time_total
+    run_usage_info["cpu_time_total_str"] = function_event.cpu_time_total_str
+    run_usage_info["cpu_time"] = function_event.cpu_time
+    run_usage_info["cpu_time_str"] = function_event.cpu_time_str
+    run_usage_info["cuda_time"] = function_event.cuda_time
+    run_usage_info["cuda_time_str"] = function_event.cuda_time_str
+    return run_usage_info
+
+
+def get_resource_usage_report(kineto_events, function_events, info):
+    """Get relevant information from Kineto events and function events exported by the profiler."""
+    mem_records = [[evt, False] for evt in kineto_events if evt.name() == "[memory]"]
+    mem_records_acc = MemRecordsAcc(mem_records)
+    main_kineto_events = sorted(
+        (evt for evt in kineto_events if "ludwig" in evt.name()), key=lambda x: x.correlation_id()
+    )
+    main_function_events = sorted((evt for evt in function_events if "ludwig" in evt.name), key=lambda x: x.id)
+    assert [evt.id for evt in main_function_events] == [evt.correlation_id() for evt in main_kineto_events]
+    assert [evt.name for evt in main_function_events] == [evt.name() for evt in main_kineto_events]
+    for kineto_event, function_event in zip(main_kineto_events, main_function_events):
+        run_usage_info = {}
+        run_usage_info = get_devices_usage(kineto_event, mem_records_acc, run_usage_info)
+        run_usage_info = get_device_timing(function_event, run_usage_info)
+        info[function_event.name]["runs"].append(run_usage_info)
+    return info
+
+
+def export_metrics_from_torch_profiler(p: profile, experiment_name: str):
+    """Export time and resource usage metrics (CPU and CUDA) from a PyTorch profiler."""
+    # events in both of these lists are in chronological order.
+    kineto_events = p.profiler.kineto_results.events()
+    function_events = p.profiler.function_events
+    assert Counter([evt.name for evt in function_events if "ludwig" in evt.name]) == Counter(
+        [evt.name() for evt in kineto_events if "ludwig" in evt.name()]
+    )
+    info = initialize_stats_dict(function_events)
+    info = get_resource_usage_report(kineto_events, function_events, info)
+    for code_block_tag, report in info.items():
+        os.makedirs(os.path.join(os.getcwd(), experiment_name, "metrics_report"), exist_ok=True)
+        file_path = os.path.join(
+            os.getcwd(), experiment_name, "metrics_report", f"{code_block_tag}_resource_usage.json"
```

Comment:
I would recommend taking the `outdir` as a separate explicit argument rather than deriving it implicitly from `os.getcwd()`. Users of this function might be in for a surprise when calling this results in results written to some directory elsewhere.